gradle/publishing.gradle

```diff
@@ -14,6 +14,63 @@
  * limitations under the License.
  */
 
+apply plugin: 'java-gradle-plugin'
+apply plugin: 'com.gradle.plugin-publish'
+//apply plugin: 'com.jfrog.bintray'
+
+ext {
+    pluginIdPrefix = 'org.asciidoctor'
+
+    pluginExtraText = (version.contains('-alpha') || version.contains('-beta')) ?
+        " (If you need a production-ready version of the AsciidoctorJ plugin for Gradle use a 1.5.x release of 'org.asciidoctor.convert' instead)"
+        : ''
+
+
+    configurePlugin = {  String providedId, String providedDisplayName, String providedDescription, List<String> providedTags ->
+        final String providedName = providedId.replaceAll(~/\./,'')
+        final File props = file("src/main/resources/META-INF/gradle-plugins/${providedId}.properties")
+        if(!props.exists()) {
+            throw new GradleException( "${props} does not exist")
+        }
+        String className
+        for( String line : props.readLines() ) {
+            def match = line =~ /^implementation-class\s*=\s*(.+?)$/
+
+            if(match.matches()) {
+                className = match[0][1]
+                break
+            }
+        }
+
+        if(className == null) {
+            throw new GradleException("${props} does not contain implemention-class")
+        }
+
+        gradlePlugin {
+            plugins {
+                "${providedName}Plugin" {
+                    id = providedId
+                    implementationClass = className
+                }
+            }
+        }
+
+        pluginBundle {
+            plugins {
+                "${providedName}Plugin" {
```

Comment:
If ti sis till required to prefix IDs with amd additional `org.asciidoctor`, this line needs to be modified,

Reply:
I think we're good with the current setup